install/gcp/bootstrap/gcp_on_prem_envoy_bootstrap.json

```diff
@@ -0,0 +1,98 @@
+{
```

Comment:
Why do we need this when we have `./install/gcp/bootstrap/gcp_envoy_bootstrap.json`?

Reply:
Thanks for the review! (I am still working this PR). 

I create a new template primarily because of the "call_credentials" of "grpc_services " needs to be set differently when the k8s cluster is not on GCP.

Reply:
Why cant you use `"google_compute_engine": {}`?

If all we are changing is the call credentials this PR can just add one more option to the existing template, I don't think we will accept yet another bootstrap template

Reply:
"google_compute_engine": {} and "GOOGLE_APPLICATION_CREDENTIALS " only works on GCE.

SG. I can update the existing template as you suggested.